config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -34,10 +41,12 @@ public ConfigServiceGrpcImpl(ConfigStore configStore) {
   public void upsertConfig(UpsertConfigRequest request,
       StreamObserver<UpsertConfigResponse> responseObserver) {
     try {
-      long configVersion =
-          configStore.writeConfig(getConfigResource(request), getUserId(), request.getConfig());
+      ConfigResource configResource = getConfigResource(request);
+      Value existingConfig = configStore.getConfig(configResource);
```

Comment:
Out of curiosity, does doc store itself support merging? Should it?

Reply:
Doc store supports updating a document but it doesn't support deep merging(the way we require it) of keys. Even if it did, I don't think we would be able to use it here as we're creating a new document for every new version.

Reply:
true